### PR TITLE
implement a way to validate class prop/methods when set/call

### PR DESCRIPTION
**Pull request by MohamedElmdary**
### Description
implement a way to validate class prop/methods when set/call

### Changes
- feat: Add a class decrator with options to validate any needed prop/method
- feat: Use Validate class decrator in ComputeCapacity class to throw error when needed
- chore: Remove skip from tests and run them using SKIP_PROPS_VALIDATION=true env var

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2821

### Documentation PR

For UI changes, Please provide the Documetation PR on [info_grid](https://github.com/threefoldtech/info_grid)

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/grid_client/src/helpers/validator.ts [hunk 1]
```diff
@@ -31,4 +31,106 @@ function validateHexSeed(seed: string, length: number): boolean {
   return true;
 }
 
-export { validateObject, validateInput, validateHexSeed };
+interface ValidationOptions {
+  props?: boolean | string | string[];
+  methods?: boolean | string | string[];
+}
+
+function Validate(options?: ValidationOptions): ClassDecorator {
```
Comment (Mahmoud-Emad) [inline packages/grid_client/src/helpers/validator.ts:39]: Can you please provide docstrings for all of the created functions and describe the purpose of each function? Also, we typically use the `validateInput` as a decorator in the grid client. I know it's different, but at the very least we need to change the name of the newly created decorator. what do you think?

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]
Good job ya Rabe3 :)

File: .github/workflows/grid_client_tests.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@ jobs:
       RMB_PROXY: true
       STORE_SECRET: secret
       MNEMONIC: ${{ secrets.MNEMONIC }}
+      SKIP_PROPS_VALIDATION: true
```
Comment (AhmedHanafy725) [inline .github/workflows/grid_client_tests.yml:26]: please remove this and update the test expectation to throw when setting the value on the object

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

**Review by Mahmoud-Emad — APPROVED** [review]

File: .github/workflows/grid_client_tests.yml (hunk 2 above)
Comment (MohamedElmdary) [reply .github/workflows/grid_client_tests.yml]: Fixed
image

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**